internal/pkg/manifest/transform.go

```diff
@@ -27,6 +27,7 @@ var defaultTransformers = []mergo.Transformers{
 	placementArgOrStringTransformer{},
 	subnetListOrArgsTransformer{},
 	unionTransformer[string, HTTPHealthCheckArgs]{},
+	unionTransformer[bool, VPCEndpoint]{},
```

Comment:
hmm do we have to update this slice every time we use a `Union` type in the manifest? This worries me a little bit.

My brain is struggling retaining all the places that we need to make a change when adding a field now to the manifest:
1. Need to add validate
2. Need to add transformer? if not, why not?
3. Need to potentially update `requiredEnvironmentFeatures` if using environment controlled parameters.

I think I'm just worried that we don't have mechanisms to catch 2. and 3. like this PR resulting in bugs.
Either we need to think of a way of simplifying the process, or we need mechanisms catching gaps like local integ tests.

Reply:
resolved in separate PR! #4189